docs/BestPractices.md

```diff
@@ -0,0 +1,123 @@
+# Best Practices for Building Data Processing Pipelines
+
+Each of our pipelines are developed using the best practices detailed below. We describe each of these best practices to give insight as to why they are important and we provide examples to give you a sense of how to apply them.
```

Comment:
I want to be a little more .. humble? here - we **aim** to follow these best practices, I am not sure if we always hit them for every pipeline.

Reply:
I updated this to "WARP pipeline development is guided by..."